components/style/dom_apis.rs

```diff
@@ -241,10 +241,11 @@ where
     }
 
     if root.as_shadow_root().is_some() {
+        debug_assert!(element.as_node().is_in_document(), "Not connected?");
```

Comment:
This change I don't understand either. Please revert or explain how it's supposed to hold.